src/query/sql/src/planner/binder/ddl/account.rs

```diff
@@ -302,23 +302,22 @@ impl Binder {
                 .await?
         };
 
+        // TODO: Only user with OWNERSHIP privilege can change user options.
         let mut user_option = user_info.option.clone();
         for option in user_options {
             user_option.apply(option);
         }
 
+        // if `must_change_password` is set, user need to change password first
+        let need_change = user_option
+            .must_change_password()
+            .cloned()
+            .unwrap_or_default();
         // None means no change to make
         let new_auth_info = if let Some(auth_option) = &auth_option {
-            // if `must_change_password` is set, user need to change password first,
-            // unless user is changing self password
-            let need_change = if user.is_none() {
-                false
-            } else {
-                user_option
-                    .must_change_password()
-                    .cloned()
-                    .unwrap_or_default()
-            };
+            // if user is changing self password, always set `need_change` as false.
+            // because after this operation, the password is changed.
+            let need_change = if user.is_none() { false } else { need_change };
```

Comment:
// None means current user
        let user_info = if user.is_none() {
            self.ctx.get_current_user()?
        }

Reply:
alter user USER() identified by 'password'

The user in plan is None.

Reply:
This logic can merge with 

`let need_change = user_option
            .must_change_password()
            .cloned()
            .unwrap_or_default();` ?

Reply:
If user not change password, we set `need_change` same as `must_change_password`. And if user change other user's password, `need_change` is also same as `must_change_password`, if user change self password, we set `need_change` as false, because the password has changed, user don't need change again.

Reply:
Ok I think I got you. Maybe can add it into comment?

Reply:
ok